app/models/order.rb

```diff
@@ -89,6 +89,25 @@ def set_initial_state
     self.state ||= :draft
   end
 
+  def is_priority?
```

Comment:
Cyclomatic compexity of this method is high ? Can we do something ?

Reply:
@namrataukirde 
It's high complexity from a code processor point of view, for the human eye it isn't. 
Trying to please code climate would just make it worse for us.
I could make a small rule engine, pseudo-code :
```javascript
const priority_rules = {
'submitted': (order) => { /*write rule */ }
};

// in the order
is_priority() {
  return priority_rules[order.state](order);
}
```
I don't think the switch case is that much of a pain, but I'm okay with changing it

Reply:
Yeah, Switch case is good here. Dont want to remove switch case here. 

If we can move last_6pm logic to some method then that can work.

But whatever we have now is more readable. So not in completely favour of moving code somewhere else.

Just wanted to know everyone's thoughts.